### ISSUE TITLE
Type already exists in PHPDoc tag: Parametrized with different classes types are reported as identical

### ISSUE THREAD

**Issue by MaXal** [JetBrains/phpstorm-psalm-plugin#65]
Code to reproduce:
```
<?php
class Foo{}
class Bar{}
/**
 * @psalm-param class-string<Foo>|class-string<Bar> $param
 */
function f($param)
{
}
```

or 
```
<?php
/**
 * @template T
 */
class Foo{}
class A{}
class B{}
/**
 * @psalm-param Foo<A>|Foo<B> $param
 */
function f($param)
{
}
```

The second type `class-string<Bar>` will be marked by _Type already exists in PHPDoc tag_ inspection as well as
`Foo<B>` although they are different from the first one.

**Comment by MaXal** [comment]
Similar example:
```
<?php

/**
 * @template T
 */
class Foo{
    /**
     * @template S
     * @psalm-param T|S $a
     */
    public function f($a){
    }
}
```
